crates/shared/src/price_estimation/competition.rs

```diff
@@ -79,11 +183,33 @@ impl PriceEstimating for RacingCompetitionPriceEstimator {
 
             // Log and collect metrics.
             let (estimator_index, result) = results.into_iter().nth(best_index).unwrap();
-            let estimator = self.inner[estimator_index].0.as_str();
-            tracing::debug!(?query, ?result, estimator, "winning price estimate");
+            let estimator = self.inner[estimator_index].0.clone();
+            tracing::debug!(
+                ?query,
+                ?result,
+                estimator = *estimator,
+                "winning price estimate"
+            );
+
+            // Collect stats for winner predictions.
+            if let (Some(competition), Ok(_)) = (&self.competition, &result) {
+                let trade = Trade::from(query);
+                if let Some(Some(prediction)) = predictions.get(&trade) {
+                    let label = match prediction.winner == estimator {
+                        true => "correct",
+                        false => "incorrect",
```

Comment:
As you mentioned, more fine grained metric would be useful. But I would add those even in this PR since we will probably get some random result for correctness (let's say 50%), and won't know what to conclude from it. We probably need to know the correctness per trade.

Reply:
> We probably need to know the correctness per trade.

Yeah, I was also thinking about this but I was not sure how well grafana would handle this as we would generate a huge amount of metrics because of all the different quote combinations.

Reply:
Added fine grained metrics in [0ffa112](https://github.com/cowprotocol/services/pull/1456/commits/0ffa112f20b102cad763c2d2b8f79a57a3e9bcc1) and in case the amounts of metrics cause problems I also added a CLI flag to control the feature in [13636e7](https://github.com/cowprotocol/services/pull/1456/commits/13636e7eab52649b5df53a86629566fb509d358b).